core/src/main/java/org/infinispan/notifications/cachelistener/NotifyHelper.java

```diff
@@ -14,54 +17,73 @@
-         }
-      } else {
-         // Notify entry event after container has been updated
-         if (created) {
-            notifier.notifyCacheEntryCreated(
-                  entry.getKey(), entry.getValue(), entry.getMetadata(), false, ctx, command);
+               functionalNotifier.notifyOnWrite(() -> EntryViews.noValue(entry.getKey()));
+            }
+         } else {
+            // Notify entry event after container has been updated
+            if (created) {
+               notifier.notifyCacheEntryCreated(
+                     entry.getKey(), entry.getValue(), entry.getMetadata(), false, ctx, command);
 
-            // A write-only command only writes and so can't 100% guarantee
-            // that an entry has been created, so only send create event
-            // when the command is read-write.
-            if (!isWriteOnly)
-               functionalNotifier.notifyOnCreate(EntryViews.readOnly(entry));
+               // A write-only command only writes and so can't 100% guarantee
+               // that an entry has been created, so only send create event
+               // when the command is read-write.
+               if (!isWriteOnly)
+                  functionalNotifier.notifyOnCreate(EntryViews.readOnly(entry));
 
-            functionalNotifier.notifyOnWrite(() -> EntryViews.readOnly(entry));
-         } else {
-            notifier.notifyCacheEntryModified(entry.getKey(), entry.getValue(), entry.getMetadata(), previousValue,
-                  previousMetadata, false, ctx, command);
+               functionalNotifier.notifyOnWrite(() -> EntryViews.readOnly(entry));
+            } else {
+               notifier.notifyCacheEntryModified(entry.getKey(), entry.getValue(), entry.getMetadata(), previousValue,
+                     previousMetadata, false, ctx, command);
 
-            // A write-only command only writes and so can't 100% guarantee
-            // that an entry has been created, so only send modify when the
-            // command is read-write.
-            if (!isWriteOnly)
-               functionalNotifier.notifyOnModify(
-                  EntryViews.readOnly(entry.getKey(), previousValue, previousMetadata),
-                  EntryViews.readOnly(entry));
+               // A write-only command only writes and so can't 100% guarantee
+               // that an entry has been created, so only send modify when the
+               // command is read-write.
+               if (!isWriteOnly)
+                  functionalNotifier.notifyOnModify(
+                        EntryViews.readOnly(entry.getKey(), previousValue, previousMetadata),
+                        EntryViews.readOnly(entry));
 
-            functionalNotifier.notifyOnWrite(() -> EntryViews.readOnly(entry));
+               functionalNotifier.notifyOnWrite(() -> EntryViews.readOnly(entry));
+            }
+         }
+      }
+   }
+
+   private static Flag extractStateTransferFlag(InvocationContext ctx, FlagAffectedCommand command) {
```

Comment:
FlagBitSets.java seems a better place to me than NotifyHelper. It's related to the flags and the utility method makes sense to me there. It might be in a specific utility class `FlagExtractor ?` if any other code related to flags extracting or check is duplicated in core.